frontend/src/pages/InsideFolder.js

```diff
@@ -1,29 +1,35 @@
-import React, { useState, Component } from "react";
-import Header from "../main-components/Header";
-import Sidebar from "../main-components/Sidebar";
-import InsideFolderContent from "../main-components/InsideFolderContent";
+import React, { useState, Component, useEffect } from 'react';
+import Header from '../main-components/Header';
+import Sidebar from '../main-components/Sidebar';
+import InsideFolderContent from '../main-components/InsideFolderContent';
+import queryString from 'query-string';
 
-function InsideFolder() {
-  // Handles mobile menu button and updates sidebar view
-  const [sideSetting, setSideSetting] = useState("f");
-  const handleClick = (e) => {
-    console.log("Clicked");
-    if (sideSetting === "f") {
-      setSideSetting("t");
-    } else {
-      setSideSetting("f");
-    }
-  };
+function InsideFolder(props) {
+	const [sideSetting, setSideSetting] = useState('f');
+	const handleClick = (e) => {
+		console.log('Clicked');
+		if (sideSetting === 'f') {
+			setSideSetting('t');
+		} else {
+			setSideSetting('f');
+		}
+	};
 
-  return (
-    <div className="App">
-      <Header id="head" handleClick={handleClick}></Header>
-      <div id="main">
-        <Sidebar bool={sideSetting}></Sidebar>
-        <InsideFolderContent></InsideFolderContent>
-      </div>
-    </div>
-  );
+	let folderKey;
+	useEffect(() => {
+		const values = queryString.parse(props.location.search);
+		folderKey = values.folderKey;
```

Comment:
useEffect runs asynchronously, so this isn't guaranteed to change folderKey.

Why do we need to use useEffect in the first place? Can't this be done in the render function?

Reply:
Noted. I went ahead and fixed it.